### PR TITLE
Use dataset in route stops e2e

### PR DESCRIPTION
**Pull request by HenrikHartiala**
This change is [Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881)

### CODE REVIEW

**Review by Leitsi — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 3 of 3 files at r3, 14 of 14 files at r4, 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881)* status: all files reviewed, 2 unresolved discussions (waiting on @HenrikHartiala)

___
*[`cypress/e2e/routeStops.cy.ts` line 338 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881#-O4Avkf8AIzRPGRpYEki:-O4Avkf94FiH-kyBuMYy:bjlt6sc) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/937250547d39d5d3906744a1a44eb0370ef0c261/cypress/e2e/routeStops.cy.ts#L338)):*
> ```XML
>       // Open timing settings modal
>       lineRouteList.routeStopListItem.openTimingSettingsForm(stopLabels[0]);
>       // Unchecking IsUsedAsTimingPointCheckbox should disable IsRegulatedTimingPointCheckbox
> ```

This case doesn't seem to be tested anymore (at least this explicitly). Should it?
___
*[`cypress/pageObjects/RouteStopListItem.ts` line 23 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881#-O4AwsnJ3JOxwFkmm2ga:-O4AwsnKBFScjA9wrA1K:bw4lh5u) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/0073aa84bd8d25efde53f442872cf7df5cd830a6/cypress/pageObjects/RouteStopListItem.ts#L23)):*
> ```XML
>   }
> 
>   getStopDropdown(stopLabel: string) {
> ```

Should these be in `StopActionsDropdown` page object instead and use the within pattern in tests?

**Review by HenrikHartiala — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881)* status: 11 of 21 files reviewed, 2 unresolved discussions (waiting on @Leitsi)

___
*[`cypress/e2e/routeStops.cy.ts` line 338 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881#-O4Avkf8AIzRPGRpYEki:-O4FEOSgDRyd6x-JSMCK:b4rfch2) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/937250547d39d5d3906744a1a44eb0370ef0c261/cypress/e2e/routeStops.cy.ts#L338)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

This case doesn't seem to be tested anymore (at least this explicitly). Should it?
</blockquote></details>

Hmm, my thought process was that it is tested  with the state of the checkbox, not particularly that the "click" is tested. So it is tested as "when IsUsedAsTimingPointCheckbox is not checked, IsRegulatedTimingPointCheckbox should be disabled". 

But I will atleast add couple of describing comments here, since now that I revisited this, it was pretty hard to follow. Or maybe I've just had too little coffee today :D Please tell what do you think of the added, I can also remove them if they are not necessary.
___
*[`cypress/pageObjects/RouteStopListItem.ts` line 23 at r4](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881#-O4AwsnJ3JOxwFkmm2ga:-O4FHmwR59iflwnHRGD1:bx49kip) ([raw file](https://github.com/HSLdevcom/jore4-ui/blob/0073aa84bd8d25efde53f442872cf7df5cd830a6/cypress/pageObjects/RouteStopListItem.ts#L23)):*
<details><summary><i>Previously, Leitsi (Tommi Leinamo) wrote…</i></summary><blockquote>

Should these be in `StopActionsDropdown` page object instead and use the within pattern in tests?
</blockquote></details>

Yes absolutely, but this was actually one of the things that _at the time_ decided to leave out from this PR (I started to do this, then aborted and just added another getStopActionsDropdown method for the time), because I got a bit overwhelmed with the changes and this one affected also hastusExport as well and..
But yeah, now I had the energy to go back with this and do the change in a separate commit.

**Review by Leitsi — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/HSLdevcom/jore4-ui/881#-:-O4e3eyaC2CHwRdh2oEU:bnfp4nl)* status: 6 of 21 files reviewed, all discussions resolved (waiting on @HenrikHartiala)
